### PR TITLE
fix: fix validation for split bill

### PR DESCRIPTION
**Pull request by koko57**
### Fixed Issues

$ https://github.com/Expensify/App/issues/40756
$ https://github.com/Expensify/App/issues/40749
PROPOSAL:

### Tests

**Merchant Error**
1. Go to FAB > Split expense > Scan.
2. Create split scan expense with a few users.
3. In the group chat, click on the split preview.
4. Click Split expense.
5. Click Split expense again.
6. Verify that there will be no error under Merchant field because Merchant is not a requirement for non-workspace expense.

**Amount Error**

1. Go to FAB > Split expense > Scan.
2. Create a split scan expense with a workspace.
3. When the expense is scanning, click on the split preview.
4. Click Split expense.
5. Verify that an error will show up for the missing amount.

- [x] Verify that no errors appear in the JS console

### Offline tests

n/a

### QA Steps

**Merchant Error**
1. Go to FAB > Split expense > Scan.
2. Create split scan expense with a few users.
3. In the group chat, click on the split preview.
4. Click Split expense.
5. Click Split expense again.
6. Verify that there will be no error under Merchant field because Merchant is not a requirement for non-workspace expense.

**Amount Error**

1. Go to FAB > Split expense > Scan.
2. Create a split scan expense with a workspace.
3. When the expense is scanning, click on the split preview.
4. Click Split expense.
5. Verify that an error will show up for the missing amount.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
        - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

having problems with building the project on android now
</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/28020445/6746fed2-ad18-4355-9e05-6d891be7103e

https://github.com/Expensify/App/assets/28020445/7e6d67a6-c01f-42f9-a11b-809aedc42398

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/28020445/ed3ac6a7-cda0-4718-b503-905c97bb89e1

https://github.com/Expensify/App/assets/28020445/b785e49d-d253-4436-8817-c4e4f38ea1c1

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/28020445/04d3cdab-b45d-404c-a4d8-aec61a08d438

https://github.com/Expensify/App/assets/28020445/b765be27-4585-45ee-a078-4e4fe0b6c647

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/28020445/4715c31e-4117-4a6f-b3a4-947be0fc7abb

https://github.com/Expensify/App/assets/28020445/1cb52609-3f9f-4155-8b8a-097185a7e59a

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/28020445/1027722c-0c68-4f27-aa53-24f247114eee

</details>

### CODE REVIEW

**Comment by mountiny** [comment]
What is your ETA for the review here?

File: src/components/MoneyRequestConfirmationList.tsx [hunk 1]
```diff
@@ -577,9 +577,8 @@ function MoneyRequestConfirmationList({
             if (selectedParticipants.length === 0) {
                 return;
             }
-            if ((isMerchantRequired && isMerchantEmpty) || (shouldDisplayFieldError && TransactionUtils.isMerchantMissing(transaction ?? null))) {
+            if (isMerchantRequired && (isMerchantEmpty || (shouldDisplayFieldError && TransactionUtils.isMerchantMissing(transaction ?? null)))) {
                 setMerchantError(true);
-                return;
```
[the remark this answers is not in the document]
Comment (koko57) [reply src/components/MoneyRequestConfirmationList.tsx:582]: fixed!

File: src/components/MoneyRequestConfirmationList.tsx [hunk 2]
```diff
@@ -322,19 +320,10 @@ function MoneyRequestConfirmationList({
     const isMerchantEmpty = !iouMerchant || iouMerchant === CONST.TRANSACTION.PARTIAL_TRANSACTION_MERCHANT;
     const isMerchantRequired = isPolicyExpenseChat && !isScanRequest && shouldShowMerchant;
 
-    const isCategoryRequired = canUseViolations && !!policy?.requiresCategory;
+    const shouldDisplayMerchantError =
+        (isMerchantRequired || (isEditingSplitBill && isScanRequest)) && (shouldDisplayFieldError || formError === 'iou.error.invalidMerchant') && isMerchantEmpty;
```
[the remark this answers is not in the document]
Comment (koko57) [reply src/components/MoneyRequestConfirmationList.tsx:324]: @Ollyws ok, you're right - removing this from the condition

[the remark this answers is not in the document]
Comment (koko57) [reply src/components/MoneyRequestConfirmationList.tsx]: @Ollyws fixed

**Review by Ollyws — APPROVED** [review]
LGTM.

**Review by robertjchen — APPROVED** [review]
Looks great, thanks!

**Pull request merged**